mac/FreeChat/FreeChatAppDelegate.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  FreeChatAppDelegate.swift
+//  FreeChat
+//
+
+import SwiftUI
+
+class FreeChatAppDelegate: NSObject, NSApplicationDelegate, ObservableObject {
+  @Environment(\.openWindow) var openWindow
+  @AppStorage("selectedModelId") private var selectedModelId: String?
+
+  func application(_ application: NSApplication, open urls: [URL]) {
+    let viewContext = PersistenceController.shared.container.viewContext
+    do {
+      let req = Model.fetchRequest()
+      req.predicate = NSPredicate(format: "name IN %@", urls.map({ $0.absoluteString }))
+      let existingModels = try viewContext.fetch(req).compactMap({ $0.url })
+
+      for url in urls {
+        guard !existingModels.contains(url) else { continue }
```

Comment:
This guard is not working correctly, re-adding the same model adds it to the list again.